src/libYARP_robotinterface/src/yarp/robotinterface/Robot.h

```diff
@@ -50,6 +50,8 @@ class YARP_robotinterface_API Robot
 
     void interrupt();
     bool enterPhase(yarp::robotinterface::ActionPhase phase);
+    bool enterPhase(yarp::robotinterface::ActionPhase phase,
+                    const yarp::dev::PolyDriverList& list);
```

Comment:
Perhaps, instead of adding an overload of `enterPhase`, you could just have one with a default value?
```c++
    bool enterPhase(yarp::robotinterface::ActionPhase phase,
                    const yarp::dev::PolyDriverList& list = yarp::dev::PolyDriverList());
```

Reply:
I initially did that, then I noticed that the style of the library to forward declare in the headers almost all the classes, and this is possible with the two overloads, but not with the function with the default value